lib/src/execution_context.dart

```diff
@@ -85,6 +85,30 @@ abstract class PostgreSQLExecutionContext {
       {Map<String, dynamic>? substitutionValues,
       bool? allowReuse,
       int? timeoutInSeconds});
+
+  /// Executes a simple query on this context.
+  ///
+  /// This method is similar to [execute] except that it'll return all the data.
+  /// That is, when the query result contains data, the method will return a
+  /// [PostgreSQLResult]. If the query does not return data, then the affected row
```

Comment:
I'm wondering: wouldn't it be better to extend `PostgreSQLResult` with `affectedRowCount`? That way we don't need to introduce yet another query type, and still preserve the full information.

Reply:
The main thing that I tried to avoid here is changing the return type of `execute` from `int` to `PostgreSQLResult` as that would be a breaking change. 

If you think it's fine to change the return type of `execute`, I could that.

Reply:
Yeah, changing `execute` would be breaking, and I would leave that to a future breaking change release. However, extending `query`'s `PostgreSQLResult` return value would be non-breaking. Would it work?

Reply:
I haven't thought about it this way... I think it'd and it'd be cleaner. 

Tis means that the `query` function will have a new optional parameters called `useSimpleQuery` which internally means "use the simple query protocol instead of the extended one".. I'll give it a try now.

Reply:
I'm genuinely clueless about this part of the code, but couldn't we use the extended protocol as is used in `query`?

Reply:
No, PostgreSQL does not allow using the Extended Query Protocol when the connection is established for a Streaming Replication Protocol. 

When using `query` during a replication mode, the database will actually throw the following error:

```
ERROR. ERROR. 08P01. extended query protocol not supported in a replication connection. postgres.c. 4749. forbidden_in_wal_sender. 
```

which is how I figured this part out... 

So the only option is to use the Simple Query Protocol which is the one used by `execute`. 

Do you think changing the query definition to the following is fine:

```dart
  Future<PostgreSQLResult> query(String fmtString,
      {Map<String, dynamic>? substitutionValues,
      bool? allowReuse,
      int? timeoutInSeconds,
      bool? useSimpleQueryProtocol});
```

I'll expand the docs for `useSimpleQueryProtocol`. Also, I think spelling out `useSimpleQueryProtocol` is better than `simpleQuery` (so no one think simple query is faster). 

Please let me know what you think.

Reply:
Thanks for the explanation!
I think the `query` with the `bool? useSimpleQueryProtocol` would seem the best compromise for now. Let's go with that! Thank you!

Reply:
Updated. I created an `_execute` method that returns `PostgreSQLResult` which can replace `execute` in the future (I added a TODO comments for that).